src/main/java/org/assertj/core/error/ShouldHaveSuperclass.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2019 the original author or authors.
+ */
+package org.assertj.core.error;
+
+import java.util.StringJoiner;
+
+/**
+ * Creates an error message indicating that an assertion that verifies that a class has a given superclass failed.
+ * 
+ * @author Stefano Cordio
+ */
+public class ShouldHaveSuperclass extends BasicErrorMessageFactory {
+
+  private static final String SHOULD_HAVE_SUPERCLASS = new StringJoiner("%n", "%n", "")
+                                                                                       .add("Expecting")
+                                                                                       .add("  <%s>")
+                                                                                       .add("to have superclass:")
+                                                                                       .add("  <%s>")
+                                                                                       .add("but had:")
+                                                                                       .add("  <%s>")
+                                                                                       .toString();
```

Comment:
I used StringJoiner instead of standard string concatenation to improve readability, but happy to revert it if it's too deviating from the standard approach.

Reply:
I don't mind at all, I think we used other implementations when the code base was java < 8

Reply:
I don't mind as long as it is formatted so that we can have a good idea of how the message will look like